packages/shared/src/hooks/streaks/useReadingStreak.ts

```diff
@@ -1,27 +1,49 @@
-import { useQuery } from '@tanstack/react-query';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useRef } from 'react';
 import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
 import { getReadingStreak, UserStreak } from '../../graphql/users';
 import { useAuthContext } from '../../contexts/AuthContext';
-import { useStreakExperiment } from './useStreakExperiment';
+import { useActions } from '../useActions';
+import { ActionType } from '../../graphql/actions';
+import useDebounce from '../useDebounce';
 
 interface UserReadingStreak {
   streak: UserStreak;
   isLoading: boolean;
-  isEnabled: boolean;
+  shouldShowPopup: boolean;
+  checkReadingStreak: () => Promise<void>;
 }
 
 export const useReadingStreak = (): UserReadingStreak => {
   const { user } = useAuthContext();
-  const { shouldShowStreak } = useStreakExperiment();
   const { data: streak, isLoading } = useQuery(
     generateQueryKey(RequestKey.UserStreak, user),
     getReadingStreak,
-    { enabled: shouldShowStreak, staleTime: StaleTime.Default },
+    { enabled: true, staleTime: StaleTime.Default },
   );
+  const { checkHasCompleted } = useActions();
+  const hasReadToday =
+    new Date(streak?.lastViewAt).getDate() === new Date().getDate();
```

Comment:
I think for this comparison, we are using date-fns is same-day function. But maybe only on the backend side? 🤔

Reply:
I found this exact comparison used somewhere else for streaks, that's why took the same implementation to not have 2 different evaluations.

Reply:
https://github.com/dailydotdev/apps/blob/21de818eb6e2b94bf673dbf14b86bcd7c94cf3a4/packages/shared/src/components/streak/ReadingStreakButton.tsx#L71-#L72